zebra/zebra_vxlan.c

```diff
@@ -4636,6 +4639,21 @@ static int zl3vni_remote_rmac_add(zebra_l3vni_t *zl3vni, struct ethaddr *rmac,
 		hook_call(zebra_rmac_update, zrmac, zl3vni, false,
 			  "new RMAC added");
 
+		/* install rmac in kernel */
+		zl3vni_rmac_install(zl3vni, zrmac);
+	} else if (!IPV4_ADDR_SAME(&zrmac->fwd_info.r_vtep_ip,
+				   &vtep_ip->ipaddr_v4)) {
+		if (IS_ZEBRA_DEBUG_VXLAN)
+			zlog_debug(
+				"L3VNI %u Remote VTEP change(%s -> %s) for RMAC %s, prefix %s",
+				zl3vni->vni,
+				inet_ntoa(zrmac->fwd_info.r_vtep_ip),
+				ipaddr2str(vtep_ip, buf1, sizeof(buf1)),
+				prefix_mac2str(rmac, buf, sizeof(buf)),
+				prefix2str(host_prefix, buf2, sizeof(buf2)));
+
+		zrmac->fwd_info.r_vtep_ip = vtep_ip->ipaddr_v4;
```

Comment:
we are updating the zrmac->fwd_info.r_vtep_ip here locally with new VTEP IP. But where/when does the zrmac entry in hash (zl3vni->rmac_table) gets updated with new VTEP IP ? Even though we install, the nh entry in the hash table still has old VTEP IP value. Please check.

Reply:
zl3vni->rmac_table lookup key is rmac. For existing rmac in hash table replace with new nexthop value. Again "show evpn next-hops specific nexthop" returns the new rmac value from the hash table.